### PR TITLE
[HW_13][ts_hw3 'Utility types' task_01, task_02]

### PR DESCRIPTION
**Pull request by Maximus2806**

### CODE REVIEW

File: js_hw_11/task_01.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+type EmployeeWithoutIsManager = Omit<IEmployee, 'isManager'>;
+
+// 8. Создайте тип, который будет содержать все поля из интерфейса IEmployee и сделает их неизменяемыми (readonly)
+
+type EmployeeReadOnly = Readonly<IEmployee>;
+
+// 9. Создайте тип, для массива объектов, где в ключах могут быть строки, в значениях number, string или boolean
+
+type ArrOfObj = {[key: string]: number | string | boolean}[]
```
Comment (anatoly-karpovich) [inline js_hw_11/task_01.ts]: Use Record

File: js_hw_11/task_02.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+};
+
+class MyStorage<T extends TObj>{
+    private data: T[] = [];
+    constructor(arr?: T[]){
+        if (arr) {
+            this.data.push(...arr)
+        }
+    };
+    private getNewId(objArr:T[]):number {
```
Comment (anatoly-karpovich) [inline js_hw_11/task_02.ts]:
```suggestion
    private getNewId():number {
```
No need to have argument in this method

File: js_hw_11/task_02.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        if(!("id" in obj)){
+            const id = this.getNewId(this.data);
+            const newObj = {...obj, id}
+            this.data.push(newObj as T)
+        } else {            
+            this.data.push(obj)
+        }
+    };
+    update(obj: Pick<T, "id"> & Partial<T>):void{
+        const index = this.data.findIndex((item) => item.id == obj.id);
```
Comment (anatoly-karpovich) [inline js_hw_11/task_02.ts]: Move this find index to private method, cause it is used at least in 2 methods

**Review by anatoly-karpovich — APPROVED** [review]

File: js_hw_11/task_01.ts (hunk 1 above)
Comment (Maximus2806) [reply js_hw_11/task_01.ts]: Done

File: js_hw_11/task_02.ts (hunk 2 above)
Comment (Maximus2806) [reply js_hw_11/task_02.ts]: Done

File: js_hw_11/task_02.ts (hunk 3 above)
Comment (Maximus2806) [reply js_hw_11/task_02.ts]: Done

**Review by VladimirLQA — APPROVED** [review]

**Pull request merged**